File: frontend/src/models/invitation.ts
import {Workspace} from '@models/workspace';
import {User} from '@models/user';

export interface Invitation {
  id: number;
  email: string;
  workspace?: Workspace;
  status: 'PENDING' | 'ACCEPTED' | 'REJECTED';
  createdAt: Date;
  invitee?: User;
  invitedBy?: User;
}

File: backend/src/main/java/com/thomas/gestPro/model/Invitation.java
package com.thomas.gestPro.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "t_invitations")
public class Invitation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email; // Celui de l'invité

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private InvitationStatus status = InvitationStatus.PENDING;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "workspace_id")
    private Workspace workspace;

    @ManyToOne
    @JoinColumn(name = "invited_by")
    private User invitedBy;

    @ManyToOne
    @JoinColumn(name = "invitee")
    private User invitee; // peut être null si l'utilisateur n'a pas encore créé de compte
}
